src/Providers/AuthUserProvider.tsx

```diff
@@ -118,18 +116,24 @@ const getRedirectURL = () => {
 
 const getRedirectOr = (fallback: string) => {
   const url = getRedirectURL();
-
+  const reserverdURLS = ["/login", "/session-expired"];
   if (url) {
     try {
       const redirect = new URL(url);
+      console.log(redirect);
       if (window.location.origin === redirect.origin) {
-        return redirect.pathname + redirect.search;
+        console.log(redirect);
+        return redirect.href;
       }
       console.error("Redirect does not belong to same origin.");
     } catch {
       console.error(`Invalid redirect URL: ${url}`);
     }
+    if (reserverdURLS.includes(location.pathname)) {
+      return "/";
+    }
+    return location.origin + location.pathname;
+  } else {
+    return fallback;
```

Comment:
This seems to be an unnecessary complication for a simple signout/signin with redirect.

Could you summarise why exactly we need this? (I may be wrong or didn't understand what exactly is being solved with reserved URLs blacklist)

Reply:
since now we pass location.pathname as fallback, Upon login we just end up at /login which is then showed as an invalid page, hence to mitigate this. The reserved urls are necessary, since they should always lead to "/" if no redirect is present